flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/hybrid/tiered/storage/TieredStorageConsumerClient.java

```diff
@@ -43,8 +68,25 @@ public void start() {
     }
 
     public Optional<Buffer> getNextBuffer(int subpartitionId) {
-        // TODO, the detailed logic will be completed when the memory tier is introduced..
-        return Optional.empty();
+        Optional<Buffer> buffer = Optional.empty();
+        for (TierConsumerAgent tiereConsumerAgent : tierConsumerAgents) {
+            buffer =
+                    tiereConsumerAgent.getNextBuffer(
+                            subpartitionId, subpartitionNextSegmentIds[subpartitionId]);
+            if (buffer.isPresent()) {
+                break;
+            }
+        }
```

Comment:
It's weird we have to loop over all tier agents for reading each buffer. Shouldn't the storage client remember the current tier that each channel is reading from?

Reply:
Also, the argument name seems not right. How could a consumer reads data from different subpartitions?